cirq/testing/lin_alg_utils.py

```diff
@@ -36,11 +36,35 @@ def random_superposition(dim: int) -> np.ndarray:
     return state_vector
 
 
-def random_unitary(dim: int) -> np.ndarray:
+def get_seeded_state(random_state: Union[int, np.random.RandomState]
```

Comment:
I don't think this method belongs in `cirq.testing`. The names also aren't really descriptive of what's happening.

Instead of doing this, just have methods take an `Optional[np.random.RandomState]` instead of an `int`. It doesn't really make sense to pass an integer seed into `random_unitary`, that would be like seeding `randint`. You seed the generator, not the value-generating method.

Furthermore, when no `np.random.RandomState` is provided you should not be defaulting to one seeded by `None`. This will result in every call to `random_unitary` producing a new RandomState. If they happen to be seeding from the clock by default, this could result in all random unitaries being equal over the span of a millisecond. You should instead be falling back to unseeded methods.

Basically, the details of how the randomness are seeded have to be handled on a method by method basis, so you can't extract out the logic like this.

Reply:
I see that @mpharrigan asked for the seed+state double duty because it matches what happens in scikit and pandas. I guess that's fine. But it is important to not constant produce new generators when getting None; that's not what happens in the other libraries I think.

Reply:
Done, thanks for the explanation! How's this?

Reply:
@Strilanc 's concerns are good ones. I don't think the latest changes address the crux of it though. We want to use the global state if an existing state is omitted. Like this person:

https://stackoverflow.com/questions/41985484/how-to-access-numpy-default-global-random-number-generator

You would think this would be easier....

Unlike the poster, we probably won't have to worry about pickling these functions so you can probably get away with

```python
if random_state is None:
  random_state = np.random
```

or you can use the answer of pulling out `random_state = numpy.random.random.__self__`.

I don't know which is less objectionable in the eyes of the python gods.

Reply:
Gotcha. Messing with __self__ looks a bit ugly, I went with the first option.